domain/application/types.go

```diff
@@ -347,6 +347,14 @@ type ExposedEndpoint struct {
 	ExposeToCIDRs set.Strings
 }
 
+// UnitMachine holds identifying information about the machine of a unit.
+type UnitMachine struct {
+	// MachineName is the name of the machine.
+	MachineName machine.Name
+	// MachineUUID is the unique identifier of the machine.
+	MachineUUID machine.UUID
+}
```

Comment:
This is a smell.

Reply:
We split this into two methods and return each one in the service layer. Please just have a state method for each one